src/lib.rs

```diff
@@ -1 +1,40 @@
+//! # Plonk Gadgets
+//! This library cointains the gadgets that the Dusk-Network protocol needs to build it's ZK-Circuits.
+//! The library **contains generic gadgets** which are used across Dusk's tech stack, all of the other
+//! gadgets used which depend on foreign types are placed on the libraries where this types are defined.
+//!
+//!
+//! ## WARNING
+//! This implementation is not audited. Use under your own responsability.
+//!
+//! ## Content
+//! This library provides:
+//!
+//! - Scalar gadgets: `is_non-zero`, `maybe_equals`, `conditionally_select_one`, `conditionally_select_zero`.
+//! - Range gadgets: `range_check`, `max_bound`.
+
+#![doc(
+    html_logo_url = "https://lh3.googleusercontent.com/SmwswGxtgIANTbDrCOn5EKcRBnVdHjmYsHYxLq2HZNXWCQ9-fZyaea-bNgdX9eR0XGSqiMFi=w128-h128-e365"
+)]
+#![doc(html_favicon_url = "https://dusk.network/lib/img/favicon-16x16.png")]
+#![allow(clippy::suspicious_arithmetic_impl)]
+// Some structs do not have AddAssign or MulAssign impl.
+#![allow(clippy::suspicious_op_assign_impl)]
+// Variables have always the same names in respect to wires.
+#![allow(clippy::many_single_char_names)]
+// Bool expr are usually easier to read with match statements.
+#![allow(clippy::match_bool)]
```

Comment:
Remove this.
First of all, there is no such code in the repo that needs this rule.

Second of all, the comment is highly opinionated, and we should comment fact not opinion.

In our code, we use both `if` and `match` statements and expressions, since `if` is a rust sugared syntax for `match`ing boolean in Rust.

Of course we have always to evaluate case by case, but unless explicitly said otherwise for matching a boolean we usually use `if` instead of `match` (both for statement and expression).

Reply:
Right. As we discussed it's removed.